core/cas-server-core-services/src/test/java/org/apereo/cas/authentication/DefaultMultifactorTriggerSelectionStrategyTests.java

```diff
@@ -84,7 +85,7 @@ public void verifyRegisteredServiceTrigger() throws Exception {
         assertThat(strategy.resolve(VALID_PROVIDERS, null, mockService(MFA_PROVIDER_ID_2), null).orElse(null), is(MFA_PROVIDER_ID_2));
 
         assertThat(strategy.resolve(VALID_PROVIDERS, null, mockService(MFA_INVALID, MFA_PROVIDER_ID_1, MFA_PROVIDER_ID_2), null).get(),
-                is(MFA_PROVIDER_ID_1));
+                anyOf(is(MFA_PROVIDER_ID_2), is(MFA_PROVIDER_1)));
```

Comment:
This doesn't look right to me. The previous test specifically wants a match on MFA_PROVIDER_ID_1. So you probably should try to preserve order for the providers. Maybe use a LinkedHashSet?